src/content/en/fundamentals/security/credential-management/retrieve-credentials.md

```diff
@@ -4,369 +4,396 @@ book_path: /web/fundamentals/_book.yaml
+
+If the `.type` is `federated`,
+the `.provider` property is a string that represents the identity provider.
+
+### Authenticate user
+
+Once you have the credential,
+run an authentication flow depending on the type of credential,
+`password` or `federated`:
+
+    }).then(c => {
+     if (c) {
+       switch (c.type) {
+         case 'password':
+           return sendRequest(c);
+           break;
+         case 'federated':
+           return gSignIn(c);
+           break;
+       }
+     } else {
+       return Promise.resolve();
+     }
+
+When the function resolves,
+check if you've received a credential object.
+If not, it means auto sign-in couldn’t happen.
+Silently dismiss the auto sign-in process.
+
+### Update UI
+
+If the authentication is successful,
+update the UI or forward the user to the personalized page:
+
+    }).then(profile => {
+     if (profile) {
+       updateUI(profile);
+     }
+
+### Don’t forget to show authentication error message
+
+One important tip: if you succeed in obtaining a credential object 
+but fail to authenticate the user, you should show an error message:
+
+        }).catch(error => {
+          showError('Sign-in Failed');
+        });
+      }
+    }
 
-If a user requires mediation, or has multiple accounts, use the account
-chooser to let the user sign-in, skipping the ordinary sign-in form.
+To avoid user confusion,
+users should see a blue toast saying “Signing in”
+at the time of getting the credential object:
+
+<div>
+  <figure>
+    <img src="imgs/auto-sign-in.png" alt="Blue toast showing user is signing in.">
+  </figure>
+</div>
+
+### Full code example
+
+    if (navigator.credentials) {
+     if (!user.isSignedIn()) {
+       navigator.credentials.get({
+         password: true,
+         federated: {
+           provider: [
+             'https://accounts.google.com'
+           ]
+         },
+         unmediated: true
+       }).then(c => {
+         if (c) {
+           switch (c.type) {
+             case 'password':
+               return sendRequest(c);
+               break;
+             case 'federated':
+               return gSignIn(c);
+               break;
+           }
+         } else {
+           return Promise.resolve();
+         }
+       }).then(profile => {
+         if (profile) {
+           updateUI(profile);
+         }
+       }).catch(error => {
+         showError('Sign-in Failed');
+       });
+     }
+    }
 
-The account chooser typically gets invoked when the user taps the
-"Sign-In" button. The user can select an account to sign-in, for example:
+## Sign-in via account chooser
+
+If a user requires mediation, or has multiple accounts,
+use the account chooser to let the user sign-in,
+skipping the ordinary sign-in form.
+
+The account chooser typically gets invoked when the user taps the "Sign-In" button.
+The user can select an account to sign-in, for example:
+
+<!--Todo: replace this is new animation.
```

Comment:
The animations weren't removed. I was thinking of adding new animations to make things more interesting. If this is something people feel is a good idea, I will follow-up in iterations. I'm not convinced it is, as animations are hard to maintain over time. They tend to get out of sync quickly, so if you have a lot of them, it becomes a recipe for maintenance issues.